numpy/core/tests/test_datetime.py

```diff
@@ -260,6 +260,21 @@ def test_datetime_array_find_type(self):
         arr = np.array([dt, dt]).astype('datetime64')
         assert_equal(arr.dtype, np.dtype('M8[us]'))
 
+    @pytest.mark.parametrize("unit", [
+    # test all date / time units and use
+    # "generic" to select generic unit
+    ("Y"), ("M"), ("W"), ("D"), ("h"), ("m"),
+    ("s"), ("ms"), ("us"), ("ns"), ("ps"),
+    ("fs"), ("as"), ("generic") ])
+    def test_timedelta_np_int_construction(self, unit):
+        # regression test for gh-7617
+        if unit != "generic":
+            assert_equal(np.timedelta64(np.int64(123), unit),
+                         np.timedelta64(123, unit))
+        else:
+            assert_equal(np.timedelta64(np.int64(123)),
+                         np.timedelta64(123))
```

Comment:
I'm not sure allowing default units to be impliedis a good idea. Is there precedent for this in `datetime64`?

Reply:
Can you clarify? `np.timedelta64(123)` already casts to the default `m8` in the master branch; the code to handle a `np.int64(123)` in the same way is just duplicating that logic for another kind of integer input--it is an actual copy-paste of code for regular Python integer handling farther up.